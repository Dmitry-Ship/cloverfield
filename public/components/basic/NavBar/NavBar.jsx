import React, { PropTypes } from 'react';

import { link, icon } from './NavBar.styl';

import { Link } from 'react-router';

import Row from '../Row';
import Icon from '../Icon';

const NavBar = ({ items, children, className, itemClass }) => (
  <Row className={`${className}`} auto={false} >
    {items.map(({ func, to, iconName, label }, i) => {
      return (
        <Link key={i} className={`${link} ${itemClass}`} onClick={func} to={to}>
          {iconName && <Icon className={icon} name={iconName}/>}
          {label || null}
        </Link>
      )
    })
  }
  {children}
  </Row>
)

export default NavBar;

NavBar.propTypes = {
   items: PropTypes.arrayOf(PropTypes.object).isRequired,
   children: PropTypes.any,
   className: PropTypes.string,
   itemClassName: PropTypes.string

}
