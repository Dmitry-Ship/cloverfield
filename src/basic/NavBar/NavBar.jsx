import React, { PropTypes } from 'react';

import { Link } from 'react-router';

import Row from '../Row';
import Icon from '../Icon';

const NavBar = ({items, children, className, itemClassName}) => (
  <Row className={`nav-bar ${className}`} auto={false} >
    {items.map((item, i) => {
      return (
        <Link key={i} className={`nav-bar__item ${itemClassName}`} onClick={item.func} to={item.to}>
          {item.icon && <Icon className='nav-bar__icon' name={item.icon}/>}
          {item.label || null}
        </Link>
      )
    })
  }
  {children}
  </Row>
)

export default NavBar;

NavBar.propTypes = {
   items: PropTypes.arrayOf(PropTypes.object).isRequired,
   children: PropTypes.any,
   className: PropTypes.string,
   itemClassName: PropTypes.string

}
