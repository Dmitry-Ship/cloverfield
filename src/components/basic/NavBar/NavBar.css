@import 'stylesheets/colors.css';
@import 'stylesheets/basic.css';
@import 'stylesheets/normalize.css';

.item {
	display: block;
	padding-right: 10px;
	padding-left: 10px;
	min-width: 55px;
	height: 55px;
	color: var(--color__icon);
	text-align: center;
	line-height: 55px;
}

.item:hover {
	border-bottom: 3px solid var(--color__accent);
	color: var(--color__accent);
}

.item:hover .icon {
	color: var(--color__accent);
}

.hamburgerButton {
	display: none;
	cursor: pointer;
}

.icon {
	margin: 10px;
	color: $icon_color;
	vertical-align: middle;
	font-size: 24px;
}
