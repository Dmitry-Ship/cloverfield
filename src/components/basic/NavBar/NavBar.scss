@value colors: "../../../styles/colors.scss";
@value secondary, tertiary from colors;

@value breakPoints: "../../../styles/breakPoints.scss";
@value mobile from breakPoints;

.link {
  color: secondary;
  text-align: center;
  // transition: color 150ms ease-in-out;
  font-weight: bold;
  &:hover {
    color: tertiary;
  }
}

.active {
  composes: link;
  color: tertiary;
}

.icon {
  vertical-align: middle;
  padding: 0;
}

.text {
  margin: 0 .4em;
  vertical-align: middle;
  @media mobile {
    display: none;
  }
}

.content {
  margin: 0 .4rem;
}
