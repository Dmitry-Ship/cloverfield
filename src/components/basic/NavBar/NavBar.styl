@value colors "../../../styles/colors.styl";
@value secondary from colors;
@value tertiary from colors;

.link
  color secondary
  text-align center
  transition: color 150ms ease-in-out
  font-weight: bold;
  &:hover
    color tertiary

.active
  composes link
  color tertiary

.icon
  vertical-align middle
  padding: 0
  font-size: 30px

.text
  margin 0 .4em
  vertical-align middle
  +tablet()
    display: none;

.content
  margin 0 .4rem
