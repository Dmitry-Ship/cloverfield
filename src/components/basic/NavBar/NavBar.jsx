import React, { PropTypes } from 'react';

import { Link } from 'react-router';

import { link, icon } from './NavBar.styl';

import Row from '../Row';
import Icon from '../Icon';

const NavBar = ({ items, children, className, itemClass }) => (
  <Row className={`${className}`} auto={false} >
    {items.map((item, i) => <NavBarItem key={i} className={itemClass} item={item} />)}
    {children}
  </Row>
);


const NavBarItem = ({ item, className }) => (
  <Link className={`${link} ${className}`} onClick={item.func} to={item.to}>
    {item.iconName && <Icon className={icon} name={item.iconName} />}
    {item.label || null}
  </Link>
);

export default NavBar;

NavBar.propTypes = {
  items: PropTypes.arrayOf(PropTypes.object).isRequired,
  children: PropTypes.any,
  className: PropTypes.string,
  itemClassName: PropTypes.string,
};
