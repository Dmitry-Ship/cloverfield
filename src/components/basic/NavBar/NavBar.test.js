import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import NavBar from './NavBar';

describe('<NavBar />', () => {
  const links = [
    { label: 'Home', iconName: 'home', to: '/' },
    { label: 'Stack', iconName: 'info_outline', to: '/about' },
  ];

  it('should render correctly', () => {
    const tree = renderer.create(
      <MemoryRouter>
        <NavBar links={links} />
      </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('render children when passed in', () => {
    const tree = renderer.create(
      <MemoryRouter>
        <NavBar links={links}>Hello World</NavBar>
      </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
