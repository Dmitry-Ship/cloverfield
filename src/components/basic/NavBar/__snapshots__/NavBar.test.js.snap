// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> render children when passed in 1`] = `
<nav
  className={null}
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexWrap": "nowrap",
      "justifyContent": "space-between",
    }
  }
>
  <a
    className="active link"
    href="/"
    onClick={[Function]}
    style={Object {}}
  >
    <div
      className="content"
    >
      <i
        className="icon material-icons"
      >
        home
      </i>
      <span
        className="text"
      >
        Home
      </span>
    </div>
  </a>
  <a
    className="link"
    href="/about"
    onClick={[Function]}
    style={undefined}
  >
    <div
      className="content"
    >
      <i
        className="icon material-icons"
      >
        info_outline
      </i>
      <span
        className="text"
      >
        Stack
      </span>
    </div>
  </a>
  Hello World
</nav>
`;

exports[`<NavBar /> should render correctly 1`] = `
<nav
  className={null}
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexWrap": "nowrap",
      "justifyContent": "space-between",
    }
  }
>
  <a
    className="active link"
    href="/"
    onClick={[Function]}
    style={Object {}}
  >
    <div
      className="content"
    >
      <i
        className="icon material-icons"
      >
        home
      </i>
      <span
        className="text"
      >
        Home
      </span>
    </div>
  </a>
  <a
    className="link"
    href="/about"
    onClick={[Function]}
    style={undefined}
  >
    <div
      className="content"
    >
      <i
        className="icon material-icons"
      >
        info_outline
      </i>
      <span
        className="text"
      >
        Stack
      </span>
    </div>
  </a>
</nav>
`;
